bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/handler/WeatherCompanyHandler.java

```diff
@@ -0,0 +1,744 @@
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.weathercompany.internal.config.WeatherCompanyConfig;
+import org.openhab.binding.weathercompany.internal.model.DayPart;
+import org.openhab.binding.weathercompany.internal.model.Forecast;
+import org.openhab.binding.weathercompany.internal.model.PwsObservations;
+import org.openhab.binding.weathercompany.internal.model.PwsObservations.Observations;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link WeatherCompanyHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * API documentation is located here
+ * - https://docs.google.com/document/d/1eKCnKXI9xnoMGRRzOL1xPCBihNV2rOet08qpE_gArAY/edit
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class WeatherCompanyHandler extends BaseThingHandler {
+    // Five-day weather forecast URL
+    private static final String BASE_FORECAST_URL = "https://api.weather.com/v3/wx/forecast/daily/5day";
+
+    // Personal Weather Station observations URL
+    private static final String BASE_PWS_URL = "https://api.weather.com/v2/pws/observations/current";
+
+    public static final int WEATHER_COMPANY_API_TIMEOUT = 10000;
+    public static final int REFRESH_JOB_INITIAL_DELAY = 3;
+
+    // Used to sanitize the API key in the URL in debug log messages
+    private static final String REPLACE_API_KEY = "XXXXXXXXXXXXXXXXXXXXX";
+
+    private final Logger logger = LoggerFactory.getLogger(WeatherCompanyHandler.class);
+
+    private final Gson gson = new GsonBuilder().serializeNulls().create();
+
+    // Provided by handler factory
+    private @NonNullByDefault({}) TimeZoneProvider timeZoneProvider;
+    private @NonNullByDefault({}) HttpClient httpClient;
+    private @NonNullByDefault({}) SystemOfUnits systemOfUnits;
+
+    // Thing configuration
+    private @NonNullByDefault({}) WeatherCompanyConfig config;
+
+    private @Nullable String forecastUrl;
+    private @Nullable String pwsUrl;
+
+    // Job to update the forecast and PWS observations
+    private @Nullable Future<?> refreshJob;
+    private int refreshInterval;
+
+    private Runnable refreshRunnable = new Runnable() {
+        @Override
+        public void run() {
+            refreshForecast();
+            refreshPwsObservations();
+        }
+    };
+
+    public WeatherCompanyHandler(Thing thing, TimeZoneProvider timeZoneProvider, HttpClient httpClient,
+            UnitProvider unitProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+        this.httpClient = httpClient;
+        this.systemOfUnits = unitProvider.getMeasurementSystem();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
AFAIK this is the default already?

Reply:
Right. This should've been OFFLINE.

Reply:
Why would you mark it OFFLINE at first, its not correct IMHO? What about just leaving it UNKNOWN until you know the status?

Reply:
Because it's not ONLINE yet? 😉 

BTW, the [archetype](https://github.com/openhab/openhab-core/blob/master/tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__Handler.java#L80) explicitly sets the status to UNKNOWN as one of the first things it does in initialization.

I can remove it completely, or leave the call to set it to UNKNOWN (which is consistent with the archetype).